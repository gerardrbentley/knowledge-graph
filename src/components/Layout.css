:root {
  /* Solarized Color Palette */
  --base03: #002b36;
  --base02: #073642;
  --base01: #586e75;
  --base00: #657b83;
  --base0: #839496;
  --base1: #93a1a1;
  --base2: #eee8d5;
  --base3: #fdf6e3;
  --yellow: #b58900;
  --orange: #cb4b16;
  --red: #dc322f;
  --magenta: #d33682;
  --violet: #6c71c4;
  --blue: #268bd2;
  --cyan: #2aa198;
  --green: #859900;
}

.theme-container {
  /* Make the light theme the default */
  --text-emphasis: var(--base01);
  --text-primary: var(--base00);
  --text-secondary: var(--base1);
  --background: var(--base3);
  --soft-background: var(--base2);
  --swap-background: var(--base03);

  --focus: var(--blue);
  --interactive: var(--magenta);

  color: var(--text-primary);
  background-color: var(--soft-background);
  width: 100%;
  height: 100%;
}

.dark-mode-checkbox:checked ~ .theme-container {
  /* Override the default theme */
  --text-emphasis: var(--base1);
  --text-primary: var(--base0);
  --text-secondary: var(--base01);
  --background: var(--base02);
  --soft-background: var(--base03);
  --swap-background: var(--base3);

  --focus: var(--blue);
  --interactive: var(--magenta);
}

.dark-mode-label {
  margin-bottom: 1em;
}

:focus,
.dark-mode-checkbox:focus ~ .theme-container .dark-mode-label {
  outline: 2px solid var(--focus);
}

.dark-mode-checkbox:checked ~ .theme-container .dark-mode-label .slider {
  background-color: var(--cyan);
}

.dark-mode-checkbox:checked ~ .theme-container .dark-mode-label .slider {
  box-shadow: 0 0 1px var(--cyan);
}

.dark-mode-checkbox:checked ~ .theme-container .dark-mode-label .slider:before {
  -webkit-transform: translateX(22px);
  -ms-transform: translateX(22px);
  transform: translateX(22px);
}

aside,
details,
figcaption,
figure,
hgroup,
nav,
section,
summary {
  display: block;
}
audio,
canvas,
video {
  display: inline-block;
}
audio:not([controls]) {
  display: none;
  height: 0;
}
[hidden] {
  display: none;
}
html {
  font-family: Arial, Helvetica, sans-serif;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  background-color: var(--base03);
}
body {
  margin: 0 auto;
}
.header {
  color: var(--text-emphasis);
  border-bottom: var(--text-emphasis);
  border-bottom-style: solid;
  margin-bottom: 0.7rem;
  box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06);
}
.main {
  width: 85%;
  max-width: 1024px;
  margin: 0 auto 2rem;
  padding: 0.7rem;
  background-color: var(--background);
  box-shadow: 10px 10px 15px rgba(0, 0, 0, 0.1), -10px 4px 6px -2px rgba(0, 0, 0, 0.05);
}
.footer {
  color: var(--text-emphasis);
  padding: 1rem 2rem;
  border-top: var(--text-emphasis);
  border-top-style: solid;
  box-shadow: 0 -4px 6px -1px rgba(0, 0, 0, 0.1);
}

a {
  color: var(--cyan);
}
a:focus {
  outline: thin dotted;
  font-size: medium;
  font-weight: 600;
  color: var(--magenta) !important;
  transition: all 0.3s;
}
a:active,
a:hover {
    outline: 0;
    color: var(--magenta) !important;
    font-weight: 600;
    transition: all 0.3s;
}
a:visited {
  color: var(--magenta);
}

h1 {
  color: var(--violet);
  margin-bottom: 0.3rem;
}
h2 {
  color: var(--blue);
}
h3,
h4,
h5,
h6 {
  color: var(--cyan);
}

hr {
  color: var(--text-emphasis);
}

table, th, td {
  padding: 4px;
  border: 1px solid var(--text-emphasis);
}


/* Rounded sliders */
.slider.round {
  border-radius: 38px;
}

.slider.round:before {
  border-radius: 50%;
}

/*
visibility-hidden utility class

Source: https://github.com/h5bp/html5-boilerplate

Hide only visually, but have it available for screen readers:
https://snook.ca/archives/html_and_css/hiding-content-for-accessibility
*/
.visually-hidden {
  position: absolute;
  overflow: hidden;
  clip: rect(0 0 0 0);
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
  white-space: nowrap; /* 1. */
}


.switch {
  position: relative;
  display: inline-block;
  width: 60px;
  height: 38px;
}

.switch input {
  opacity: 0;
  width: 0;
  height: 0;
}

.slider {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #ccc;
  -webkit-transition: .4s;
  transition: .4s;
}

.slider:before {
  position: absolute;
  content: "";
  height: 30px;
  width: 30px;
  left: 4px;
  bottom: 4px;
  background-color: white;
  -webkit-transition: .4s;
  transition: .4s;
}

/* PrismJS 1.21.0
https://prismjs.com/download.html#themes=prism-tomorrow&languages=markup+css+clike+javascript+bash+java+json+json5+python+jsx+tsx+typescript&plugins=show-language+toolbar+copy-to-clipboard */
/**
 * prism.js tomorrow night eighties for JavaScript, CoffeeScript, CSS and HTML
 * Based on https://github.com/chriskempson/tomorrow-theme
 * @author Rose Pritchard
 */

 code[class*="language-"],
 pre[class*="language-"] {
   color: #ccc;
   background: none;
   font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
   font-size: 1em;
   text-align: left;
   white-space: pre;
   word-spacing: normal;
   word-break: normal;
   word-wrap: normal;
   line-height: 1.5;
 
   -moz-tab-size: 4;
   -o-tab-size: 4;
   tab-size: 4;
 
   -webkit-hyphens: none;
   -moz-hyphens: none;
   -ms-hyphens: none;
   hyphens: none;
 
 }
 
 /* Code blocks */
 pre[class*="language-"] {
   padding: 1em;
   margin: .5em 0;
   overflow: auto;
   box-shadow: 0.05rem 0.05rem 0.5rem var(--text-secondary);
 }
 
 :not(pre) > code[class*="language-"],
 pre[class*="language-"] {
   background: #2d2d2d;
 }
 
 /* Inline code */
 :not(pre) > code[class*="language-"] {
   padding: .1em .3em;
   border-radius: .3em;
   white-space: normal;
 }
 
 .token.comment,
 .token.block-comment,
 .token.prolog,
 .token.doctype,
 .token.cdata {
   color: #999;
 }
 
 .token.punctuation {
   color: #ccc;
 }
 
 .token.tag,
 .token.attr-name,
 .token.namespace,
 .token.deleted {
   color: #e2777a;
 }
 
 .token.function-name {
   color: #6196cc;
 }
 
 .token.boolean,
 .token.number,
 .token.function {
   color: #f08d49;
 }
 
 .token.property,
 .token.class-name,
 .token.constant,
 .token.symbol {
   color: #f8c555;
 }
 
 .token.selector,
 .token.important,
 .token.atrule,
 .token.keyword,
 .token.builtin {
   color: #cc99cd;
 }
 
 .token.string,
 .token.char,
 .token.attr-value,
 .token.regex,
 .token.variable {
   color: #7ec699;
 }
 
 .token.operator,
 .token.entity,
 .token.url {
   color: #67cdcc;
 }
 
 .token.important,
 .token.bold {
   font-weight: bold;
 }
 .token.italic {
   font-style: italic;
 }
 
 .token.entity {
   cursor: help;
 }
 
 .token.inserted {
   color: green;
 }
 
 div.code-toolbar {
   position: relative;
 }
 
 div.code-toolbar > .toolbar {
   position: absolute;
   top: .3em;
   right: .2em;
   transition: opacity 0.3s ease-in-out;
   opacity: 0;
 }
 
 div.code-toolbar:hover > .toolbar {
   opacity: 1;
 }
 
 /* Separate line b/c rules are thrown out if selector is invalid.
    IE11 and old Edge versions don't support :focus-within. */
 div.code-toolbar:focus-within > .toolbar {
   opacity: 1;
 }
 
 div.code-toolbar > .toolbar .toolbar-item {
   display: inline-block;
 }
 
 div.code-toolbar > .toolbar a {
   cursor: pointer;
 }
 
 div.code-toolbar > .toolbar button {
   background: none;
   border: 0;
   color: inherit;
   font: inherit;
   line-height: normal;
   overflow: visible;
   padding: 0;
   -webkit-user-select: none; /* for button */
   -moz-user-select: none;
   -ms-user-select: none;
 }
 
 div.code-toolbar > .toolbar a,
 div.code-toolbar > .toolbar button,
 div.code-toolbar > .toolbar span {
   color: #bbb;
   font-size: .8em;
   padding: 0 .5em;
   background: #f5f2f0;
   background: rgba(224, 224, 224, 0.2);
   box-shadow: 0 2px 0 0 rgba(0,0,0,0.2);
   border-radius: .5em;
 }
 
 div.code-toolbar > .toolbar a:hover,
 div.code-toolbar > .toolbar a:focus,
 div.code-toolbar > .toolbar button:hover,
 div.code-toolbar > .toolbar button:focus,
 div.code-toolbar > .toolbar span:hover,
 div.code-toolbar > .toolbar span:focus {
   color: inherit;
   text-decoration: none;
 }
 
 