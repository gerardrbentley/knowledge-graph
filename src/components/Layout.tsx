import * as React from 'react'
import { Link, useStaticQuery, graphql } from 'gatsby'
import { MarkdownRemark } from '../types'

export interface LayoutProps {
    next?: MarkdownRemark
    prev?: MarkdownRemark

}

const Layout: React.FunctionComponent<LayoutProps> = ({ next, prev, children }) => {
    const data = useStaticQuery(
        graphql`
            query {
                site {
                    siteMetadata {
                        title
                        description
                        author
                        year
                    }
                }
            }
        `
    )
    console.log({data})
    const { title, description, author, year } = data.site.siteMetadata
    let header = (
        <div style={{
            display: 'flex',
            alignItems: 'center',
            justifyContent: "center",
            fontFamily: 'sans-serif',
        }}>
            <div style={{
                display: 'flex',
                flexDirection: 'column',
                alignItems: 'center',
                marginLeft: '1rem',
                marginRight: '1rem',
                width: '30%'
            }}>
                <div style={{
                    marginBottom: '1rem'
                }}>
                    <Link to='/'>Index Page</Link>
                </div>
                <div>
                    {prev && <Link to={prev.frontmatter.path}>Prev ({prev.frontmatter.title})</Link>}
                </div>
            </div>
            <div style={{
                display: 'flex',
                flexDirection: 'column',
                alignItems: 'center',
                textAlign: 'center',
                marginLeft: '1rem',
                marginRight: '1rem',
                width: '40%'
            }}>
                <h2 style={{ marginBottom: '.3rem' }}>{title}</h2>
                <p style={{ opacity: 0.8 }}>{description}</p>
            </div>
            <div style={{
                display: 'flex',
                flexDirection: 'column',
                alignItems: 'center',
                marginLeft: '1rem',
                marginRight: '1rem',
                width: '30%'
            }}>
                <div style={{
                    marginBottom: '1rem'
                }}>
                    <a href="/#">Other Link</a>
                </div>
                <div>
                    {next && <Link to={next.frontmatter.path}>Next ({next.frontmatter.title})</Link>}
                </div>
            </div>
        </div>
    )

    let footer = (
        <div style={{
            display: 'flex',
            alignItems: 'center',
            justifyContent: "center",
            fontFamily: 'sans-serif',
            marginBottom: '3rem'
        }}>
            <div>
                {prev && <Link to={prev.frontmatter.path}>Prev ({prev.frontmatter.title})</Link>}
            </div>
            <div style={{
                display: 'flex',
                flexDirection: 'column',
                alignItems: 'center',
                marginLeft: '1rem',
                marginRight: '1rem',
                width: '40%',
                textAlign: 'center'
            }}>
                <span>
                    Â© {year}, Built by {author} Using <a href="https://www.gatsbyjs.org">GatsbyJS</a>
                </span>
            </div>
            <div>
                {next && <Link to={next.frontmatter.path}>Next ({next.frontmatter.title})</Link>}
            </div>
        </div>
    )

    return (
        <div
            style={{
                marginLeft: 'auto',
                marginRight: 'auto',
                maxWidth: '768px',
            }}
        >
            <header>
                {header}
            </header>
            <hr></hr>
            <main style={{
                display: 'flex',
                flexDirection: 'column',
                alignItems: 'center',
                marginLeft: 'auto',
                marginRight: 'auto',
                width: '85%',
            }}>
                {children}
            </main>
            <hr></hr>
            <footer>
                {footer}
            </footer>
        </div>
    )
}

export default Layout