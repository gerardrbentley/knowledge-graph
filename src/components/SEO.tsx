import * as React from 'react'
import Helmet from 'react-helmet'
import { useStaticQuery, graphql } from 'gatsby'
import { SiteMetadataQuery } from '../types'


interface PureSEOProps {
    sitedata: SiteMetadataQuery
    description?: string
    lang?: string
    meta?: []
    title: string
}

interface SEOProps {
    description?: string
    lang?: string
    meta?: []
    title: string
}

export const PureSEO: React.FunctionComponent<PureSEOProps> = ({sitedata, description, lang, meta, title}) => {

    const metaDescription = description || sitedata.site.siteMetadata.description

    return (
        <Helmet
            htmlAttributes={{
                lang,
            }}
            title={title}
            titleTemplate={`%s | ${sitedata.site.siteMetadata.title}`}
            meta={[
                {
                    name: `description`,
                    content: metaDescription,
                },
                {
                    property: `og:title`,
                    content: title,
                },
                {
                    property: `og:description`,
                    content: metaDescription,
                },
                {
                    property: `og:type`,
                    content: `website`,
                },
                {
                    name: `twitter:card`,
                    content: `summary`,
                },
                {
                    name: `twitter:creator`,
                    content: sitedata.site.siteMetadata.author,
                },
                {
                    name: `twitter:title`,
                    content: title,
                },
                {
                    name: `twitter:description`,
                    content: metaDescription,
                },
            ].concat(meta || [])}
            />
    )
}

const SEO: React.FunctionComponent<SEOProps> = (props) => {
    const site = useStaticQuery(
        graphql`
            query {
                site {
                    siteMetadata {
                        title
                        description
                        author
                    }
                }
            }
        `
    )

    return <PureSEO {...props} sitedata={ site }/>
}

export default SEO