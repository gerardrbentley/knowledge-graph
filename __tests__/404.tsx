import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import NotFoundPage from '../src/pages/404'
import Layout from '../src/components/Layout'
import SEO from '../src/components/SEO'

const renderer = new ShallowRenderer()

describe("404 Page", () => {
    it("renders successfully", () => {
        renderer.render(<NotFoundPage></NotFoundPage>)
        const tree = renderer.getRenderOutput()
        expect(tree).toMatchSnapshot()
    }),
    it("renders components", () => {
        renderer.render(<NotFoundPage></NotFoundPage>)
        const tree = renderer.getRenderOutput()
        expect(tree.type).toBe(Layout)
        expect(tree.props.children).toContainEqual(<SEO title="404: Not Found" />)
        expect(tree.props.children).toContainEqual(<h1>Not Found</h1>)
    })

})