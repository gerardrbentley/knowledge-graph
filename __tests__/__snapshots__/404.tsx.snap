// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page renders successfully 1`] = `
<Layout>
  <SEO
    title="404: Not Found"
  />
  <h1>
    Not Found
  </h1>
  <p>
    Sorry, the link you followed can't be found
  </p>
</Layout>
`;
